src/metrics.rs

```diff
@@ -2951,6 +2987,33 @@ impl GooseAttack {
                 errors_template = "".to_string();
             }
 
+            let graph_rps_template: String;
+            if self.configuration.requests_per_second {
+                let mut aggregated_count: HashMap<usize, usize> = HashMap::new();
```

Comment:
I wonder if instead we could use the logger thread to efficiently write time-data to log files (I assume we'd need multiple: one for each type of chart we add to the reports). So instead of buckets, in the main parent loop we'd track the total # of requests, total # of tasks, total # of users seen each second and push that to the logger thread. Then, when we write the reports we could perhaps read in the time-data from these files instead of trying to hold them in memory? In this way we always have access to per-second granularity, but we can decide how often to plot a point (every second, every 10 seconds, every minute, etc) depending on how long the load test actually ran, something we can't know at start-time and run-time), and we can even add controls allow us to zoom in/out on the data.

Reply:
I changed the approach and now it stores # of requests for each second as you described here. 

I calculated how much data that approach uses in memory and I got about ~350kB per day (4 bytes/second * 86400 = 345600 bytes). Considering how little memory this uses I am not sure if it is worth adding the complexity with storing data to files. Thoughts?

Reply:
Completely agreed, there's no reason to add this sort of complexity for such small data sizes -- even if we add many charts.